s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java

```diff
@@ -190,23 +171,23 @@
                                                            @PathVariable("institution") String institutionSlug)
             throws NotFoundException {
         if (request.getGroupSlugs() != null) {
-            val user = appUserService.findByEmail(request.getEmail())
-                    .orElseThrow(() -> new NotFoundException("User not found for email: '" + request.getEmail() + "'"));
-            groupService.updateUsersGroups(user.getId(), institutionSlug, request.getGroupSlugs());
+            groupService.updateUserGroups(request, institutionSlug);
         }
         return ResponseEntity.ok().build();
     }
 
     @Operation(summary = "Get user profile")
     @ApiResponses({
             @ApiResponse(responseCode = "200", description = "User was retrieved"),
-            @ApiResponse(responseCode = "403", description = "Forbidden: Don't have permission to get profile")
+            @ApiResponse(responseCode = "403", description = "Forbidden: Don't have permission to get profile"),
+            @ApiResponse(responseCode = "404", description = "Forbidden: Don't have permission to get profile")
```

Comment:
Don't have permission? Rather user not found, however, it shouldn't happen. I would omit the 404 code here.

Reply:
msg changed, we throw notfound, so i chose to let it be the way it was :)